ios/FluentUI/PersonaButton/PersonaButton.swift

```diff
@@ -51,29 +51,44 @@ import SwiftUI
 
     /// Secondary text to be displayed under the persona image (e.g. last name or email address).
     var secondaryText: String? { get set }
+
+    /// Design token set for this control, to use in place of the control's default Fluent tokens.
+    var overrideTokens: PersonaButtonTokens? { get set }
 }
 
 /// View that represents a persona button.
-public struct PersonaButton: View {
-    @Environment(\.theme) var theme: FluentUIStyle
-    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
-    @Environment(\.sizeCategory) var sizeCategory: ContentSizeCategory
-    @ObservedObject var tokens: MSFPersonaButtonTokens
-    @ObservedObject var state: MSFPersonaButtonStateImpl
-
+public struct PersonaButton: View, TokenizedControlInternal {
     /// Creates a new PersonaButton instance.
     /// - Parameters:
     ///   - size: The MSFPersonaButtonSize value used by the PersonaButton.
     public init(size: MSFPersonaButtonSize) {
         let state = MSFPersonaButtonStateImpl(size: size)
         self.state = state
-        self.tokens = state.tokens
     }
 
-    internal init(state: MSFPersonaButtonStateImpl, action: (() -> Void)?) {
+    public var body: some View {
+        let action = state.onTapAction ?? {}
+        SwiftUI.Button(action: action) {
+            VStack(spacing: 0) {
+                avatarView
+                personaText
+                Spacer(minLength: tokens.verticalPadding)
+            }
+        }
+        .frame(minWidth: adjustedWidth, maxWidth: adjustedWidth, minHeight: 0, maxHeight: .infinity)
+        .background(Color(dynamicColor: tokens.backgroundColor))
+        .resolveTokens(self)
+        .resolveTokenModifier(self, value: state.buttonSize)
```

Comment:
Another nit on naming that is not applicable directly to the PersonaButton:

We don't usually see the term "Modifier" in the func names.
What do you think about renaming these to "**resolvedTokens**" and "**resolvedTokensForControl**"?

Reply:
Fair point about not using `Modifier` since it doesn't mean the same thing in this context as a SwiftUI modifier.

The goal for `resolvedTokensModifier` is that we should be observing the value passed in as a second argument. 

Honestly they can probably both be named `resolveTokens` since they have different arguments and thus different signatures (`resolveTokens(_:)` vs `resolveTokens(_:value:)`).